cast/src/lib.rs

```diff
@@ -66,9 +67,17 @@ impl Network {
     }
 }
 
-pub fn get_provider(url: &str) -> Result<JsonRpcClient<HttpTransport>> {
+pub fn get_provider(url: &str, network: &Network) -> Result<JsonRpcClient<HttpTransport>> {
     let parsed_url = Url::parse(url)?;
     let provider = JsonRpcClient::new(HttpTransport::new(parsed_url));
+
+    let rt = Runtime::new()?;
+    let provider_chain_id = rt.block_on(provider.chain_id())?;
+    let cli_chain_id = network.get_chain_id();
```

Comment:
I've started wondering if we could get rid of the `network` parameter entirely.
Since we are calling network for its chain_id here we do not need `network` parameter anymore.
It's useful only when getting an account but we could change the logic of accessing the network's value.
What do you think?

Reply:
I think UX would be improved because it's one parameter less :)

Reply:
hmm, makes sense. I guess we should still get network from scarb.toml though, right? in order to compare the networks and throw an error when there is mismatch, to avoid accidental declare/deploy to wrong network?